### DISCUSSION TITLE
How should API to hide all columns with ignoreIds be used?

### DISCUSSION THREAD

**Discussion by ashgard071** [nocodb/nocodb#4533] [Q&A]
Hi, 

I would like to show/hide columns for a view from API. My idea was to use https://all-apis.nocodb.com/#tag/DB-view/operation/db-view-hide-all-column for this purpose. I'm using postman to test the behaviour. The POST URL I'm using looks as follows: 
`
http://localhost:8080/api/v1/db/meta/views/vw_li4y8j1bxihg2l/hide-all?ignoreIds=["cl_xvb5gbf4iwuhzy"]
`
I've added the `xc-token` to the header for authentication.

The response I get is:
```
{
    "msg": "update `nc_grid_view_columns_v2` set `show` = false, `updated_at` = CURRENT_TIMESTAMP where `fk_view_id` = 'vw_li4y8j1bxihg2l' and not (`fk_column_id` in '[\\\"cl_xvb5gbf4iwuhzy\\\"]') - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''[\\\"cl_xvb5gbf4iwuhzy\\\"]')' at line 1"
}
```

I check the documentation (https://all-apis.nocodb.com/#tag/DB-view/operation/db-view-hide-all-column) if my call is correct, here is stated that `ignoreIds` should be an Array of arrays.
Then I checked the code https://github.com/nocodb/nocodb/blob/develop/packages/nocodb/src/lib/meta/api/viewApis.ts#L91:
```
async function hideAllColumns(req: Request<any, any>, res) {
  res.json(
    await View.hideAllColumns(
      req.params.viewId,
      <string[]>(req.query?.ignoreIds || [])
    )
  );
}
```
which gives me the impression that a list of strings should be used (which I did). 

This is confirmed by the implementation of View.hideAllColumns (https://github.com/nocodb/nocodb/blob/c1dec6d2922cb4f4a9273e12e21969a5e66f5274/packages/nocodb/src/lib/models/View.ts#L1056) 

My question is: how should the API call look like for using this endpoint?

Thanks

**Comment by o1lab** [comment]
You can check what are the API inputs from network tab in the browser.

**Reply by ashgard071** [reply]
Good suggestion. I noticed a different endpoint is being used by the UI: https://all-apis.nocodb.com/#tag/DB-view-column/operation/db-view-column-update. This works for me so problem solved. Although I'm not sure if the ignoreIds feature is working for the issue in the main thread.
